libraries/common/process.effekt

```diff
@@ -0,0 +1,9 @@
+module process
+
+extern io def exit(errorCode: Int): Nothing =
+  js   "(function() { process.exit(${errorCode}) })()"
+  llvm """
+    call void @exit(i64 %errorCode)
+    ret %Pos zeroinitializer
+  """
+  chez "(exit ${errorCode})"
```

Comment:
There's no MLton version, because I haven't figured out what to do there:
- I found that [`OS.Process.exit OS.Process.failure/success` could be used](https://stackoverflow.com/a/47849043), but it fails on my machine:
```
$ effekt --backend=ml libraries/common/process.effekt
Warning: <bogus> ~1.~1.
  Type of variable was not inferred and could not be generalized: it.
    type: ???
    in: val it = main_2612 (fn a => a)
```
- ... and I can't see a way to return a custom exit code at all.

Reply:
How do we define `exit` for `jsWeb`? #454

Reply:
Regarding ML and exit, in #551 we try to enforce that main returns `unit`. So the type of `it` should be Unit (not sure how to annotate this here, though).

Reply:
Funnily enough, the warning is really just a warning & now I have a version which (on a POSIX system) allows returning any POSIX return code. :)